### PR TITLE
Development monitoring chatflow

### PR DESCRIPTION
**Pull request by ranatrk**
https://github.com/threefoldtech/home/issues/761
- Add new monitoring chatflow to deploy prometheus, grafana, and a redis container 
- Add monitoring chatflow solution schema
- Create grafana package to add location to be used in admin dashboard
- Add the monitoring chatflow to the admin dashboard
- Add optional logs to redis in the following chatflows: *Gitea*, *Minio*, *Ubuntu*, *Generic flist(your_flist)*, *monitoring*(prometheus and grafana)

### CODE REVIEW

File: ThreeBotPackages/zerobot/admin/frontend_src/sources/views/main.js [hunk 1]
```diff
@@ -118,6 +118,9 @@ export default class TopView extends JetView {
             }, {
                 id: "deployed_gateway_4to6",
                 value: '<span><img class="solutions-icon" src="static/img/ip.png"/>4 to 6 Gateway</span>'
+            }, {
+                id: "deployed_monitoring",
+                value: '<span><img class="solutions-icon" src="static/img/redis.png"/>Monitoring</span>'
```
Comment (zaibon) [inline ThreeBotPackages/zerobot/admin/frontend_src/sources/views/main.js:123]: Can we pick something else then the redis logo as a icon for the monitoring chat flow. Redis has 0 connotation with monitoring this is confusing.

File: ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+    @j.baseclasses.chatflow_step(title="Access keys")
+    def public_key_get(self):
+        self.user_form_data["Public key"] = self.upload_file(
+            """Please add your public ssh key, this will allow you to access the deployed containers using ssh.
+                    Just upload the file with the key""",
+            required=True,
+        ).split("\n")[0]
+
+    @j.baseclasses.chatflow_step(title="Prometheus container resources")
+    def prometheus_container_resources(self):
```
Comment (zaibon) [inline ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py:63]: Are the proposed default value for CPU and memory viable for a Prometheus container ? 1GB seems very very few for a db like that...

File: ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+    @j.baseclasses.chatflow_step(title="Prometheus volume details")
+    def prometheus_volume_details(self):
+        form = self.new_form()
+        vol_disk_type = form.drop_down_choice(
+            "Please choose the type of disk for the volume to be attached to the Prometheus container",
+            ["SSD", "HDD"],
+            required=True,
+            default="SSD",
+        )
+        vol_disk_size = form.int_ask("Please specify the volume size", required=True, default=10)
```
Comment (zaibon) [inline ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py:96]: Add in which unit the size of the volume is given

File: ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+        self.user_form_data["Prometheus Volume Disk type"] = vol_disk_type.value
+        self.user_form_data["Prometheus Volume Size"] = vol_disk_size.value
+
+    @j.baseclasses.chatflow_step(title="Grafana container resources")
+    def grafana_container_resources(self):
+        form = self.new_form()
+        cpu = form.int_ask(
+            "Please add how many CPU cores are needed for the Grafana container", default=1, required=True
+        )
+        memory = form.int_ask("Please add the amount of memory in MB", default=1024, required=True)
```
Comment (zaibon) [inline ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py:107]: Are the proposed default value for CPU and memory viable for a grafana container ?

File: ThreeBotPackages/zerobot/admin/frontend_src/sources/views/main.js (hunk 1 above)
Comment (ranatrk) [reply ThreeBotPackages/zerobot/admin/frontend_src/sources/views/main.js:123]: Sure, I can look for something more generic like a simple graph?

File: ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py (hunk 2 above)
Comment (ranatrk) [reply ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py:63]: is 4GB reasonable enough as a default?

File: ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+            self.resv_id, self.user_form_data["Solution name"], "tfgrid.solutions.monitoring.1", self.user_form_data
+        )
+
+    @j.baseclasses.chatflow_step(title="Success", disable_previous=True)
+    def success(self):
+        res = f"""\
+            # Your containers have been deployed successfully. Your reservation id is: {self.resv_id}
+            ## Prometheus
+            #### Access container by ```ssh root@{self.ip_addresses["Prometheus"]}``` where you can manually customize the solutions you want to monitor
+            #### Access Prometheus UI through ```{self.ip_addresses["Prometheus"]}/9090/graph``` which is accessed through your browser
```
Comment (zaibon) [inline ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py:330]:
```suggestion
            #### Access Prometheus UI through ```{self.ip_addresses["Prometheus"]}:9090/graph``` which is accessed through your browser
```

**Comment by zaibon** [comment]
2020-06-11_3153x1585_scrot
I think the detail of the deployed solution is missing most of the information

File: ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py (hunk 4 above)
Comment (ranatrk) [reply ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py:107]: Yes and verified with @muhamadazmy as well.

File: ThreeBotPackages/zerobot/admin/frontend_src/sources/views/main.js (hunk 1 above)
Comment (zaibon) [reply ThreeBotPackages/zerobot/admin/frontend_src/sources/views/main.js:123]: sounds good

File: ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py (hunk 2 above)
Comment (zaibon) [reply ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py:63]: well I don't know exactly, But I guess we can find recommendation on their website.

File: ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py (hunk 4 above)
Comment (zaibon) [reply ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py:107]: Ok scratch there, seems 1GiB is more then enough.

from https://grafana.com/docs/grafana/latest/installation/requirements/
> Grafana does not use a lot of resources and is very lightweight in use of memory and CPU.
Minimum recommended memory: 255 MB Minimum recommended CPU: 1

File: ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py (hunk 2 above)
Comment (ranatrk) [reply ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py:63]: According to this https://prometheus.io/docs/prometheus/1.8/storage/#memory-usage 3Gib would work as a default

**Comment by ranatrk** [comment]
> I think the detail of the deployed solution is missing most of the information

With the metadata of the reservation changed, it will now include these items:
Screenshot from 2020-06-11 20-18-57

Screenshot from 2020-06-11 20-19-06

File: ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py (hunk 5 above)
Comment (ranatrk) [reply ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py]: updated here https://github.com/threefoldtech/jumpscaleX_threebot/pull/764/commits/2c57640372d6d533176a56676396ab87810b0f3b

File: ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py (hunk 3 above)
Comment (ranatrk) [reply ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py]: updated here https://github.com/threefoldtech/jumpscaleX_threebot/pull/764/commits/2c57640372d6d533176a56676396ab87810b0f3b#diff-0e24746aa7feb62e0d805614631f5de0R96

File: ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py (hunk 2 above)
Comment (ranatrk) [reply ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py:63]: updated here https://github.com/threefoldtech/jumpscaleX_threebot/pull/764/commits/2c57640372d6d533176a56676396ab87810b0f3b#diff-0e24746aa7feb62e0d805614631f5de0R68

File: ThreeBotPackages/zerobot/admin/frontend_src/sources/views/main.js (hunk 1 above)
Comment (ranatrk) [reply ThreeBotPackages/zerobot/admin/frontend_src/sources/views/main.js]: updated to this https://github.com/threefoldtech/jumpscaleX_threebot/blob/2c57640372d6d533176a56676396ab87810b0f3b/ThreeBotPackages/zerobot/admin/frontend_src/static/img/monitoring_graph.png

**Pull request merged**